### ISSUE TITLE
Framework question: how the alpha be updated? Some different between the code and paper.

### ISSUE THREAD

**Issue by awecefil** [quark0/darts#138]
Hi, @quark0 
Thank you for this great work!
However, I have some questions about the code and the paper.

In the paper, one of the advantages of One-shot model and Cell-based search is the reduction of computation cost. And I thought Cell-based search is like this:

**1.  Search for best cell(Normal cell and Reduction cell) separately at first.
2.Stack these 2 cells to construct the entire model by human and then re-train it.**

So the whole process will be in 2 stage(search for cells and stack them with re-train the model).

But when I read the code, I found that the entire model(stack by cells) has been defined at first, so the training loss and validation loss is come from the entire model.
So, **how to calculate the gradient of the architecture weight(alpha) and then backprop it?** 
Because when update the alpha, there are many cells with different number of weights(filters) due to different channel size but alpha are shared by same kind of cell,

If there are somethings I missing, please tell me, thanks.
